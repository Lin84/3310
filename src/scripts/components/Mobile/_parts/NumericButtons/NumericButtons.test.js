import React from 'react';
import renderer from 'react-test-renderer';

// react component
import NumericButtons from './index';

// props:
const handleClick = () => {};
const numericButtonsDefinition = [
    { label: '1 _', value: '' },
    { label: '2 abc', value: '2' },
    { label: '3 def', value: '3' },
    { label: '4 ghi', value: '4' },
    { label: '5 jkl', value: '5' },
    { label: '6 mno', value: '6' },
    { label: '7 pqrs', value: '7' },
    { label: '8 tuv', value: '8' },
    { label: '9 wxyz', value: '9' }
];

it('renders correctly', () => {
    const tree = renderer
        .create(<NumericButtons
            handleClick={handleClick}
            definitions={numericButtonsDefinition}
        />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
