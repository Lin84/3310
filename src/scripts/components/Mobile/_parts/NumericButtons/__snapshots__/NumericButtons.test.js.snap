// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <button
    className="btn btn mobile__numberic-button"
    onClick={[Function]}
  >
    1 _
  </button>,
  <button
    className="btn btn mobile__numberic-button"
    onClick={[Function]}
  >
    2 abc
  </button>,
  <button
    className="btn btn mobile__numberic-button"
    onClick={[Function]}
  >
    3 def
  </button>,
  <button
    className="btn btn mobile__numberic-button"
    onClick={[Function]}
  >
    4 ghi
  </button>,
  <button
    className="btn btn mobile__numberic-button"
    onClick={[Function]}
  >
    5 jkl
  </button>,
  <button
    className="btn btn mobile__numberic-button"
    onClick={[Function]}
  >
    6 mno
  </button>,
  <button
    className="btn btn mobile__numberic-button"
    onClick={[Function]}
  >
    7 pqrs
  </button>,
  <button
    className="btn btn mobile__numberic-button"
    onClick={[Function]}
  >
    8 tuv
  </button>,
  <button
    className="btn btn mobile__numberic-button"
    onClick={[Function]}
  >
    9 wxyz
  </button>,
]
`;
